flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/mock/MockKeyedStateBackend.java

```diff
@@ -221,9 +226,11 @@ public void notifyCheckpointAborted(long checkpointId) {
             @Nonnull CheckpointOptions checkpointOptions) {
         return new FutureTask<>(
                 () ->
-                        SnapshotResult.of(
-                                new MockKeyedStateHandle<>(
-                                        copy(stateValues, stateSnapshotFilters))));
+                        emptySnapshot
```

Comment:
why we must introduce a `emptySnapshot` here?

Reply:
A non-empty snapshot can't be handled by JM because  `MockKeyedStateHandle` can't be serialized by MetadataSerializer. Adding a branch to (de)serialize mock state handle to production code doesn't seem right to me, so I decided to have an empty snapshot instead.